routing_table.go

```diff
@@ -206,6 +206,12 @@ func (r *routingTable) cleanup(cleanupPeriod time.Duration, p *peerStore) (needP
 			r.kill(n, p)
 			continue
 		}
+                // kill old and currently unused nodes if nodeCount is > maxNodes
+                if len(r.addresses) > p.maxNodes && time.Since(n.createTime) > cleanupPeriod && len(n.pendingQueries) == 0 {
```

Comment:
I'd say we should kill the node even if there are pending queries. If it's so old, better refresh the routing table with newer models?

Reply:
we may have just sent out a query a second ago to that specific node, because of the nearest distance to the searched hash.. so we don't want to loose that result?